src/bootstrap/build/sanity.rs

```diff
@@ -49,12 +54,29 @@ pub fn check(build: &mut Build) {
         }
         need_cmd("cmake".as_ref());
         if build.config.ninja {
-            need_cmd("ninja".as_ref())
+            need_cmd("ninja".as_ref());
         }
         break
     }
 
-    need_cmd("python".as_ref());
+    let python = need_cmd("python".as_ref());
+
+    // On OSX we can only use the python from /usr/bin because otherwise it's
+    // incompatible with the LLDB python framework we load during the LLDB
+    // tests. It's somewhat common for a homebrew-installed python to exist, but
+    // we're unfortunately incompatible with that.
+    if build.config.build.ends_with("apple-darwin") {
+        if !python.starts_with("/usr/bin") {
+            panic!("
+
+Python detected ({}) is not /usr/bin/python and is unlikely to be able
+to run lldb tests. If this Python is installed via Homebrew then you'll likely
+need to uninstall it entirely (or move it out of 'PATH') to get lldb tests
+working.
+
+", python.display());
```

Comment:
I unfortunately didn't get that far in testing. If I had Homebrew Python installed and `/usr/local/bin/python` in my PATH then I'd still get weird errors about a missing "six" module, but once `/usr/local/bin` was removed from PATH that error went away. 

Python would successfully report an import of the `lldb` module (or at least the interpreter would exit cleanly) but would print some other errors during the `./configure` step.

I can try to get a clearer picture of what's going on soon as well.